src/did/src/lib.rs

```diff
@@ -121,7 +113,97 @@ pub struct Poll {
     pub end_timestamp_secs: u64,
 }
 
-impl From<PollCreateData> for Poll {
+impl PendingPoll {
+    /// Returns the total number of votes.
+    pub fn total_votes(&self) -> u64 {
+        (self.no_voters.len() + self.yes_voters.len()) as u64
+    }
+
+    /// Returns the number of yes votes.
+    pub fn yes_votes(&self) -> u64 {
+        self.yes_voters.len() as u64
+    }
+
+    /// Returns the number of no votes.
+    pub fn no_votes(&self) -> u64 {
+        self.no_voters.len() as u64
+    }
+
+    /// Closes the poll
+    pub fn close(self, result: PollResult) -> ClosedPoll {
```

Comment:
Wouldn't it be better to derive the result of the poll here instead of giving it from outside? The current approach opens a way to create a closed poll with more `no` votes, that is still accepted. Of course, it will probably not happen, but it's usually better to have type system protect you from creating an invalid instance.

It's not a blocker, so I will approve the PR, but consider refactoring this.

Reply:
The PollResult is provided from outside because here we don't know the poll-approval policy. As of today, we require yes-votes to be more than no-votes, but we could apply different strategies like "All yes-votes", "At least 60% yes-votes" and so on. If needed, this can be refactored in the future to accept an "ApprovalStrategy" instead of a PollResult